browser/app/js/objects/ObjectItem.js

```diff
@@ -35,30 +34,29 @@ export const ObjectItem = ({
   onClick
 }) => {
   return (
-    <div 
-      className={
-        "fesl-row " + 
-        classNames({
-          "fesl-row-selected": checked
-        })
-      }
-      data-type={getDataType(name, contentType)}
-    >
+    <div className={"fesl-row"} data-type={getDataType(name, contentType)}>
       <div className="fesl-item fesl-item-icon">
         <div className="fi-select">
           <input
             type="checkbox"
             name={name}
+            checked={checked}
+            onChange={() => {
+              checked ? uncheckObject(name) : checkObject(name)
+            }}
           />
           <i className="fis-icon" />
+          <i className="fis-helper" />
         </div>
       </div>
       <div className="fesl-item fesl-item-name">
         <a
           href="#"
           onClick={e => {
             e.preventDefault()
-            checked ? uncheckObject(name) : checkObject(name)
+            if (getDataType(name, contentType) === "folder") {
+              onClick()
+            }
```

Comment:
onClick prop is not passed to this component in case of Object, so we could just do
```javascript
if(onClick) {
    onClick()
}
```